packages/components/src/menu/types.ts

```diff
@@ -173,73 +197,137 @@ export interface MenuItemProps {
+	 * attribute accordingly, enabling support for all elements, including those
+	 * that don't support the native `disabled` attribute.
+	 *
+	 * @default false
+	 */
+	disabled?: Ariakit.MenuItemCheckboxProps[ 'disabled' ];
+	/**
+	 * Allows the component to be rendered as a different HTML element or React
+	 * component. The value can be a React element or a function that takes in the
+	 * original component props and gives back a React element with the props
+	 * merged.
+	 */
+	render?: Ariakit.MenuItemCheckboxProps[ 'render' ];
 	/**
 	 * The checkbox menu item's name.
 	 */
-	name: string;
+	name: Ariakit.MenuItemCheckboxProps[ 'name' ];
 	/**
 	 * The checkbox item's value, useful when using multiple checkbox menu items
 	 * associated to the same `name`.
 	 */
-	value?: string;
+	value?: Ariakit.MenuItemCheckboxProps[ 'value' ];
 	/**
 	 * The controlled checked state of the checkbox menu item.
+	 *
+	 * Note: this prop will override the `defaultChecked` prop.
 	 */
-	checked?: boolean;
+	checked?: Ariakit.MenuItemCheckboxProps[ 'checked' ];
 	/**
 	 * The checked state of the checkbox menu item when it is initially rendered.
 	 * Use when not wanting to control its checked state.
+	 *
+	 * Note: this prop will be overriden by the `checked` prop, if it is defined.
 	 */
-	defaultChecked?: boolean;
+	defaultChecked?: Ariakit.MenuItemCheckboxProps[ 'defaultChecked' ];
 	/**
-	 * Event handler called when the checked state of the checkbox menu item changes.
+	 * A function that is called when the checkbox's checked state changes.
 	 */
-	onChange?: ( event: React.ChangeEvent< HTMLInputElement > ) => void;
+	onChange?: Ariakit.MenuItemCheckboxProps[ 'onChange' ];
 }
 
-export interface MenuRadioItemProps
-	extends Omit< MenuItemProps, 'prefix' | 'hideOnClick' > {
+export interface MenuRadioItemProps {
+	/**
+	 * The contents of the menu item, which could include one instance of the
+	 * `Menu.ItemLabel` component and/or one instance of the `Menu.ItemHelpText`
+	 * component.
+	 */
+	children: Ariakit.MenuItemRadioProps[ 'children' ];
+	/**
+	 * The contents of the menu item's suffix, such as a keyboard shortcut.
+	 */
+	suffix?: React.ReactNode;
+	/**
+	 * Determines if the menu should hide when this item is clicked.
+	 *
+	 * **Note**: This behavior isn't triggered if this menu item is rendered as a
+	 * link and modifier keys are used to either open the link in a new tab or
+	 * download it.
+	 *
+	 * @default false
+	 */
+	hideOnClick?: Ariakit.MenuItemRadioProps[ 'hideOnClick' ];
 	/**
-	 * Whether to hide the menu popover when the menu item is clicked.
+	 * Determines if the element is disabled. This sets the `aria-disabled`
+	 * attribute accordingly, enabling support for all elements, including those
+	 * that don't support the native `disabled` attribute.
 	 *
 	 * @default false
 	 */
-	hideOnClick?: boolean;
+	disabled?: Ariakit.MenuItemRadioProps[ 'disabled' ];
```

Comment:
Added an explicit `false` value for the `disabled` prop — this is also reflected in the code when destructuring props